### PR TITLE
Add Google Bare Metal Solution as a cloud platform

### PR DESCRIPTION
**Pull request by alex-basinov**
Adds [Google Bare Metal Solution](https://cloud.google.com/bare-metal) as an allow-listed cloud platform. 
Please let me know if there's anything else I need to do to make this change properly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ release.
 
 ### Semantic Conventions
 
+- Add GCP Bare Metal Solution as a cloud platform.
```
Comment (joaopgrassi) [inline CHANGELOG.md:14]:
```suggestion
- Add GCP Bare Metal Solution as a cloud platform
 ([#64](https://github.com/open-telemetry/semantic-conventions/pull/64))
```
We always add the PR link also to the changelog entry.

**Review by jsuereth — APPROVED** [review]
Approved barring two things:

1. Can you sign the CLA ?
2. Joao's suggestion on the changelog.

**Review by arminru — APPROVED** [review]

**Comment by alex-basinov** [comment]
1. I am covered by a corporate CLA now. How can I tell EasyCLA bot to re-run the check ? /easycla comment doesn't seem to be doing anything.
2. suggestion applied.

**Pull request merged**